examples/hosts/app-integration/schema-upgrade/src/migrator/migrator.ts

```diff
@@ -60,6 +73,12 @@ export class Migrator extends TypedEventEmitter<IMigratorEvents> implements IMig
      * that a freshly-loaded migrated container is in collaborating state.
      */
     private readonly takeAppropriateActionForCurrentMigratable = () => {
+        if (!this.connected) {
+            // If we are not connected we should wait until we reconnect and try again.
+            this.currentModel.migrationTool.once("connected", this.takeAppropriateActionForCurrentMigratable);
+            return;
+        }
+
         const migrationState = this._currentModel.migrationTool.migrationState;
         if (migrationState === "migrating") {
```

Comment:
If this is the state machine, some comments would be helpful what each state should be representing. "migrating" can mean "im migrating" or "someone is migrating". Some comments around the "else" case as well.

Reply:
There are comments in `migrationInterfaces/migrationTool.ts` which explain the migration states. Do you think we should bring it to migrator.ts as well to increase visibility?

Reply:
" * * migrating - a proposal to migrate has been accepted.  The data is currently being migrated.
I could not understand what it meant in this particular context.